src/main/java/io/pakland/mdas/githubstats/infrastructure/github/repository/CommitGitHubRepository.java

```diff
@@ -24,22 +27,24 @@ public CommitGitHubRepository(WebClientConfiguration webClientConfiguration) {
     }
 
     @Override
-    public List<Commit> fetchCommitsFromPullRequest(FetchCommitsFromPullRequestRequest request)
+    public List<Commit> fetchCommitsFromPullRequestByPage(PullRequest pullRequest, Integer page)
         throws HttpException {
-        String query = String.format("/repos/%s/%s/pulls/%s/commits?%s",
-            request.getRepositoryOwner(),
-            request.getRepositoryName(), request.getPullRequestNumber(),
-            getRequestParams(request));
-        List<Commit> maybeResult = cache.get(query);
-        if (maybeResult != null) {
-            return maybeResult;
-        }
-
         try {
-            logger.info(" - Fetching commits for pull request: " + request.getPullRequestNumber()
-                .toString());
+            final String uri = String.format("/repos/%s/%s/pulls/%s/reviews?%s",
+                pullRequest.getRepository().getOwnerLogin(),
+                pullRequest.getRepository().getName(),
+                pullRequest.getNumber(),
+                new GitHubPageableRequest(page, 100).getRequestUriWithParameters()
+            );
+
+            List<Commit> maybeResult = cache.get(uri);
+            if (maybeResult != null) {
+                return maybeResult;
+            }
```

Comment:
Why is this inside the `try catch` block? It can not throw any exception that is catched by that block which is why it was before the `try catch`.

Also, in the `CommentGitHubRepository` you have kept it outise the `try catch`.

Reply:
I think I got it wrong in `CommitGithubRepository` as the uri string now needs to build a GithubPageableRequest which i've made to throw an exception if it's constructed with a page number below 0